tensorflow/lite/kernels/embedding_lookup_test.cc

```diff
@@ -116,6 +114,33 @@ TEST(EmbeddingLookupOpTest, SimpleTest) {
               })));
 }
 
+TEST(EmbeddingLookupOpTest, IndexTooHighTest) {
+  EmbeddingLookupOpModel m({3}, {3, 2, 4});
+  m.SetInput({3, 0, 2});
+  m.Set3DWeightMatrix<float>(
+      [](int i, int j, int k) { return i + j / 10.0f + k / 100.0f; });
+
+  EXPECT_DEATH(m.Invoke(), "Embedding Lookup: index out of bounds.");
+}
+
+TEST(EmbeddingLookupOpTest, NegativeIndexTest) {
+  EmbeddingLookupOpModel m({3}, {3, 2, 4});
+  m.SetInput({-3, 0, 2});
+  m.Set3DWeightMatrix<float>(
+      [](int i, int j, int k) { return i + j / 10.0f + k / 100.0f; });
+
+  EXPECT_DEATH(m.Invoke(), "Embedding Lookup: index out of bounds.");
+}
+
+TEST(EmbeddingLookupOpTest, SingleIndexTest) {
+  EmbeddingLookupOpModel m({1}, {3, 2, 4});
```

Comment:
The main behavior being tested here isn't the "SIngleIndexTest". It's the "IndexTooHigh" behavior.

Reply:
There is only one behavior we want to test (failing on the index being too high). This is currently being tested in both "IndexTooHighTest" and "SingleIndexTest". We should dedup this.

Reply:
@alanchiao thanks for the feedback, i have updated based on your comments.

Regards
Amit